public/apps/configuration/panels/role-edit.tsx

```diff
@@ -35,13 +43,69 @@ interface RoleEditDeps extends AppDependencies {
   sourceRoleName: string;
 }
 
+type OptionSeletion = EuiComboBoxOptionOption[];
+
+function buildPermissionOptions(optionsList: string[]) {
+  return optionsList.map(e => ({ label: e }));
+}
+
 export function RoleEdit(props: RoleEditDeps) {
+  const [roleName, setRoleName] = useState('');
+  const [roleClusterPermission, setRoleClusterPermission] = useState<OptionSeletion>([]);
+  useEffect(() => {
+    const action = props.action;
+    if (action == 'edit' || action == 'duplicate') {
+      const fetchData = async () => {
+        try {
+          const roleData = await getRoleDetail(props.coreStart.http, props.sourceRoleName);
+          setRoleClusterPermission(buildPermissionOptions(roleData.cluster_permissions));
+
+          if (action == 'edit') {
+            setRoleName(props.sourceRoleName);
+          }
+          if (action == 'duplicate') {
+            setRoleName(props.sourceRoleName + '_copy');
+          }
+        } catch (e) {
+          console.log(e);
```

Comment:
add a todo here to promote user friendly message which they know action to take. maybe get wording from ux/pm